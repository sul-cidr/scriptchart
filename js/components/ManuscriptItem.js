import React from "react";

class ManuscriptItem extends React.Component {
  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }
  handleClick() {
    if (this.props.hasOwnProperty("onHiddenChange")) {
      this.props.onHiddenChange("show", "column", this.props.manuscriptID);
    }
  }

  render() {
    return (
      <li onClick={this.handleClick} style={{ cursor: 'pointer' }}>{this.props.display}</li>
    );
  }
}

export default ManuscriptItem;