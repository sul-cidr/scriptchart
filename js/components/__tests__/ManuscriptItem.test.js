/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import ManuscriptItem from '../ManuscriptItem';


describe('ManuscriptItem test', () => {
  it('ManuscriptItem should match snapshot', () => {
    const component = renderer.create(<ManuscriptItem
       />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
