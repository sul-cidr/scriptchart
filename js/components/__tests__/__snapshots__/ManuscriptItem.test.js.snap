// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManuscriptItem test ManuscriptItem should match snapshot 1`] = `
<li
  onClick={[Function]}
  style={
    Object {
      "cursor": "pointer",
    }
  }
/>
`;
